bpf/bpf_sock.c

```diff
@@ -273,27 +276,60 @@ static __always_inline int __sock4_xlate(struct bpf_sock_addr *ctx,
 	if (sock4_skip_xlate(svc, in_hostns, ctx->user_ip4))
 		return -EPERM;
 
-	key.slave = (sock_local_cookie(ctx_full) % svc->count) + 1;
+#if defined(ENABLE_SESSION_AFFINITY) && defined(BPF_HAVE_NETNS_COOKIE)
+	/* Session affinity id (a netns cookie) */
+	__u64 client_id = 0;
```

Comment:
should only be scoped inside the if body?

Reply:
Yes, it should be scoped inside the if body. Or do you mean adding extra `{ ... }` to scope it?